### PR TITLE
coins: add a --print flag that can print the accumulated append-list

### PR DESCRIPTION
**Pull request by ckoparkar**

### CODE REVIEW

File: examples/src/coins/coins.sml [hunk 1]
```diff
@@ -1,67 +1,77 @@
...
 
-fun lenA (ls : AList) : int =
+fun lenA (ls : 'a AList) : int =
   case ls of
     ANil => 0
   | ASing _ => 1
   | Append (l, r) => lenA l + lenA r
 
-fun append (ls1 : AList) (ls2 : AList) : AList =
+fun showA (f : 'a -> string) (ls : 'a AList) : string =
```
Comment (ckoparkar) [inline examples/src/coins/coins.sml:19]: @shwestrick I'm not aware of a way with which we can avoid writing this and the `show_list` function by hand. Is there?

**Comment by ckoparkar** [comment]
Sample output:

```
% ./bin/coins '@MLton' number-processors 1 -- -N 5 -repeat 3 --sequential --print
N 5
sequential? true
repeat 3
Sequential: 2. Finished in: 0.0000s.
Sequential: 2. Finished in: 0.0000s.
Sequential: 2. Finished in: 0.0000s.
AList=(Append (ASing [5]) (ASing [1,1,1,1,1]))
```

File: examples/src/coins/coins.sml (hunk 1 above)
Comment (shwestrick) [reply examples/src/coins/coins.sml:19]: Yeah, unfortunately we have to write them by hand. For `show_list`, a slightly nicer way might be:
```
fun show_list f xs =
  "[" ^ String.concatWith "," (List.map f xs) ^ "]"
```

Comment (ckoparkar) [reply examples/src/coins/coins.sml:19]: Yep, that's much better. I updated the commit.

**Comment by shwestrick** [comment]
Looks great!

**Pull request merged**